Add ability to preventDefault

anubisthejackle opened the pull request:
This PR adds the ability to pass a boolean value to Keypress to run the preventDefault method on the keyup event. This will allow for situations where you want to use this feature on longer pages with the up arrow or down arrow keys without allowing it to scroll.

anubisthejackle committed:
Add ability to preventDefault

This PR adds the ability to pass a boolean value to Keypress to run the preventDefault method on the keyup event. This will allow for situations where you want to use this feature on longer pages with the up arrow or down arrow keys without allowing it to scroll.

lupas commented on src/index.vue:
```diff
@@ -8,6 +8,10 @@ export default {
       default: "keyup"
+    },
+    preventDefault: {
+      type: Bolean,
```
Could you change this to `Boolean`?

anubisthejackle replied:
Whoops! Missed that typo.

lupas commented on src/index.vue:
```diff
@@ -8,6 +8,10 @@ export default {
+    },
+    preventDefault: {
+      type: Bolean,
+      default: null
```
This line can & should be deleted.

Boolean types are by default `false` if not defined, so we can omit this line completely.

lupas reviewed (changes requested):
Hey @anubisthejacke

Thanks for the PR! :)

Could you:
- Fix the issues I marked in the PR
- Add the new Prop to the documentation in the Readme?

Thanks a lot!

anubisthejackle committed:
Update index.vue

Fix typo of Boolean, remove the default value of preventDefault

anubisthejackle committed:
Update README.md

anubisthejackle commented:
@lupas Should be good now. Let me know if you need any more changes.

lupas commented:
> @lupas Should be good now. Let me know if you need any more changes.

Hey @anubisthejackle 
Sorry, had a very busy last few days.

1. So I tested it today and it did **not** work for me.
E.g. when setting the preventDefault flag and key-code=40, I could still scroll down the page with the down arrow. Did you test your changes yourself? I tested it on Google Chrome.

2. I think we would want to do write it this way anyway:
```js
    emitEvent(event) { 
      if (event.keyCode === this.keyCode || !this.keyCode) {
        if (this.preventDefault) {
            event.preventDefault();
      }
      this.$emit("pressed", event.keyCode);
      }
    }
```
Otherwise we set preventDefault on any button's click, not just the one we want to handle.

Let me know what you think :)

anubisthejackle committed:
Update index.vue

Update position of prevent default to be contained inside the check for the correct keycode.

anubisthejackle commented:
@lupas You're absolutely right, that's a much better positioning.

To see it stop the scroll, you will need to test the preventDefault on key_down_ because that's when the scroll action happens.

lupas commented:
@anubisthejackle 
Awesome, **keydown** did the trick! ;-)

Thanks a lot, this is great!

lupas commented:
Published with [v1.2.0](https://github.com/lupas/vue-keypress/releases/tag/v1.2.0)
